src/java/org/apache/cassandra/index/sai/utils/AbortedOperationException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.utils;
+
+
+/**
+ * This exception indicates that a request was aborted, normally because it was taking too much time.
+ *
+ * It is handled in a special way by the verb handlers and the request execute method: it is simply
+ * passed to the onAborted callback without logging any message. Therefore if any logging is required,
+ * it is up to the code raising this exception to log anything.
+ */
+public class AbortedOperationException extends RuntimeException
+{
+    public AbortedOperationException()
+    {
+        super();
```

Comment:
Should we have `QueryContext` feed what information it *does* have to this so we can construct a basic diagnostic message. (We could even get the CQL from the `ReadCommand`, which seems to be readily available at context creation time...)

Reply:
As above.